src/Sarif.UnitTests/Writers/ResultLogJsonWriterTests.cs

```diff
@@ -135,5 +158,207 @@ public void ResultLogJsonWriter_MultipleDisposeAllowed()
+                                        Line = 10
+                                    },
+
+                                    new StackFrame
+                                    {
+                                        Module = "a.dll",
+                                        FullyQualifiedLogicalName = "N1.N2.C.M2",
+                                        Line = 6
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        };
+
+        private const string SerializedNotification =
+@"        {
+          ""id"": ""NOT0001"",
+          ""ruleId"": ""TST0001"",
+          ""message"": ""This is a test"",
+          ""level"": ""error"",
+          ""time"": ""2016-04-29T00:00:00.00Z"",
+          ""exception"": {
+            ""type"": ""System.AggregateException"",
+            ""message"": ""Bad thing"",
+            ""innerExceptions"": [
+              {
+                ""type"": ""System.ArgumentNullException"",
+                ""message"": ""x cannot be null"",
+                ""stack"": {
+                  ""frames"": [
+                    {
+                      ""line"": 10,
+                      ""module"": ""a.dll"",
+                      ""fullyQualifiedLogicalName"": ""N1.N2.C.M1""
+                    },
+                    {
+                      ""line"": 6,
+                      ""module"": ""a.dll"",
+                      ""fullyQualifiedLogicalName"": ""N1.N2.C.M2""
+                    }
+                  ]
+                }
+              }
+            ]
+          }
+        }";
+
+        [TestMethod]
+        public void ResultLogJsonWriter_WritesConfigurationNotifications()
+        {
+            string expected =
+@"{
+  ""version"": """ + SchemaVersion + @""",
+  ""runs"": [
+    {
+      ""tool"": {
+        ""name"": null
+      },
+      ""configurationNotifications"": [
+" + SerializedNotification + @"
+      ]
+    }
+  ]
+}";
+            string actual = GetJson(uut =>
+            {
+                uut.WriteTool(s_defaultTool);
+                uut.WriteConfigurationNotifications(s_notifications);
+            });
+            Assert.AreEqual(expected, actual);
+        }
+
+        [TestMethod]
+        public void ResultLogJsonWriter_WritesToolNotifications()
+        {
+            string expected =
+@"{
+  ""version"": """ + SchemaVersion + @""",
+  ""runs"": [
+    {
+      ""tool"": {
+        ""name"": null
+      },
+      ""toolNotifications"": [
+" + SerializedNotification + @"
+      ]
+    }
+  ]
+}";
+            string actual = GetJson(uut =>
+            {
+                uut.WriteTool(s_defaultTool);
+                uut.WriteToolNotifications(s_notifications);
+            });
+            Assert.AreEqual(expected, actual);
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(InvalidOperationException))]
```

Comment:
These two tests found a bug (I wasn't setting the `Conditions` flag after writing out the notifications). The same bug existed for `invocation`. Fixed that too and added a test for it, above.